configmanager.go

```
@@ -86,31 +101,29 @@ func (e *EnvironmentConfigManager) fetchConfig(retrying bool) error {
 		if err = e.setConfigFromResponse(resp); err != nil {
 			return err
 		}
-		break
+		return nil
 	case statusCode == http.StatusNotModified:
-		break
+		return nil
 	case statusCode == http.StatusForbidden:
-		e.pollingStop <- true
+		e.stopPolling()
 		return errorf("invalid SDK key. Aborting config polling")
 	case statusCode >= 500:
 		// Retryable Errors. Continue polling.
-		if !retrying {
-			warnf("Retrying config fetch. Status:" + resp.Status)
-			return e.fetchConfig(true)
-		}
 		warnf("Config fetch failed. Status:" + resp.Status)
```

Do we want to return an error when we hit too many 500s? Or just keep polling in the background and returning default values?

---

I tried to preserve the old behaviour of continuing to poll in the background on 500, even after we've exhausted the initial retries, and NOT returning the error.

I'm not sure that's a good idea though, if we've actually broken something it'll just leave the SDK silently defaulting all variables.